docs/understanding-the-flutterflow-ui/projects/how-to-run-test-projects.md

```diff
@@ -0,0 +1,94 @@
+---
+slug: how-to-run-test-projects
+title: How to Run and Test Projects
+tags: []
+toc_max_heading_level: 5
+---
+There are 4 ways to test your project in FlutterFlow.
+- Preview
+- Test
+- Run
+- Local Run
+
+## Preview Mode
+You can use the Preview Mode to quickly try out your app on a virtual device without waiting for it to build. This is helpful primarily for navigation and animations. You can also preview your app in the Dark/Light mode and visualize it on various mobile, tablet, and desktop devices.
+
+<div class="alert alert--info" role="alert">
+#### When to Use Preview Mode
+The main benefit is that your app will load instantly. However, because of the limitations this feature is used much less often that the other testing modes.
+</div>
+<div class="alert alert--warning" role="alert">
+  #### Preview Mode Limitations
+  - Actions may not trigger or work properly.
+  - FontAwesome icons jump around when mouse hovers over certain material widgets.
+  - Firestore data is not loaded from Firebase.
+  - Firebase auth flow can't be tested. We always allow log in.
+  - API Calls can't be run or tested here.
+  - Refresh if animation actions are not working.
+  - Refresh if Clear TextFields actions are not working
+  - RevenueCat data is not loaded.
+  - Paywall actions execute as if the entitlement is active.
+  - Hero Animation may not work on dynamically generated widgets.
+  - Dropdown disabling does not work in
+  - Preview Mode.
+  - Tooltip does not work for some screen sizes in Preview Mode.
+</div>
+
+## Test Mode
```

Comment:
Can we mention somewhere under Test Mode that it basically runs a web version of the Flutter application which is why there are certain limitations (mentioned later)

Reply:
Hmm, I'm reticent to include that as I wonder if people would read into it that a Flutter app deployed to the web has a bunch of limitations. Some of the limitations are because of that (e.g. copy to clipboard) but other things are due to our cloud run configuration. IDK, what do you think? Thoughts @leighajarett?

Reply:
Yes, but this can be two different points in that case. Like "this runs web version" and "currently WE have certain limitations on our test mode that is ...." so that way the implication is not that Flutter web has these limitations. Anyway being honest about the limitations is also going to be helpful to developers to set the correct expectations.

Reply:
Sounds good. I add some copy to the end of the first paragraph and as in introduction in the caution callout.